src/property.c

```diff
@@ -234,6 +236,104 @@ void setter_background_image(Context* context, const char* key, const char* valu
+  pcre2_code* code = pcre2_compile(
+    SCHEME_LIST,
+    PCRE2_ZERO_TERMINATED,
+    PCRE2_ANCHORED | PCRE2_CASELESS | PCRE2_ENDANCHORED,
+    &errorcode,
+    &erroroffset,
+    NULL
+  );
+  if (!code) {
+    g_warning("pcre2_compile failed for errorcode `%d` at offset `%d`\n", errorcode, (int)erroroffset);
+    return;
+  }
+
+  // repetitivelly get all schemes in the list, one by one.
+  GSList* schemes = NULL;
+  int scheme_length_sum = 0;
+  while (value[0]) {
+    pcre2_match_data* match_data = pcre2_match_data_create_from_pattern(code, NULL);
+    int res = pcre2_match(
+        code,
+        value,
+        PCRE2_ZERO_TERMINATED,
+        0,
+        PCRE2_ANCHORED | PCRE2_ENDANCHORED | PCRE2_NOTEMPTY,
+        match_data,
+        NULL
+    );
+
+    if (res <= 0) {
+        switch (res) {
+        case 0:
+            printf("Ovector was not big enough. This should not happen.");
+            break;
+        case PCRE2_ERROR_NOMATCH:
+            printf("No match\n");
+            break;
+        default:
+            printf("PCRE2 match error %d\n", res);
+            break;
+        }
+        pcre2_match_data_free(match_data);
+        pcre2_code_free(code);
+        return;
+    }
+
+    PCRE2_SIZE* ovector = pcre2_get_ovector_pointer(match_data);
+    int length = ovector[3] - ovector[2];
+    if (length > 0) {
+        schemes = g_slist_prepend(schemes, g_strndup(value + ovector[2], length)); // get first scheme
+        scheme_length_sum += length + 1; // 1 for separater `|` or terminal null char
+    }
+
+    pcre2_match_data_free(match_data);
+    value = &value[ovector[3] + 1]; // move pointer forward by one word
+  }
+  pcre2_code_free(code);
+
+  // if no schemes specified, remove current regex and return immediately
+  if (scheme_length_sum == 0) {
+    if (context->layout.uri_tag >= 0) {
+      vte_terminal_match_remove(context->layout.vte, context->layout.uri_tag);
+    }
+    return;
+  }
+
+  gchar scheme_pattern[scheme_length_sum];
+  gchar* p = scheme_pattern;
+  for (GSList* scheme = schemes; scheme; scheme = scheme->next) {
+    p = g_stpcpy(p, scheme->data);
+    *p = '|';
+    ++p;
+  }
+  scheme_pattern[scheme_length_sum - 1] = '\0'; // replace last `|` with null char
+  gchar* uri_pattern = g_strconcat("(?:", scheme_pattern, ")", SCHEMELESS_URI, NULL);
+  g_slist_free(schemes);
+
+  GError* error = NULL;
+  VteRegex* regex = vte_regex_new_for_match(uri_pattern, -1, PCRE2_UTF | PCRE2_MULTILINE | PCRE2_CASELESS, &error);
+  g_free(uri_pattern);
+
+  if (error) {
+    g_warning("Error when parsing css: %s", error->message);
+    g_error_free(error);
+  } else {
+    if (context->layout.uri_tag >= 0) {
+      vte_terminal_match_remove(context->layout.vte, context->layout.uri_tag);
+    }
+    int tag = vte_terminal_match_add_regex(context->layout.vte, regex, 0);
+    context->layout.uri_tag = tag;
+    vte_terminal_match_set_cursor_name(context->layout.vte, tag, "hand");
+    vte_regex_unref(regex);
+  }
```

Comment:
If using setter, `config_set_str()` should be called manually in setter to store the value.

Reply:
Is that necessary? There are several property setters like `setter_cjk_width` or `setter_role` which do not do `config_set_str` to store the values. Could you tell me when to, and when not to call `config_set_str`?

Reply:
The problem isn't that the value is not stored but the value is not accessible by `tym.get()`. `cjk_width` and `role` are defined getter so we can get the values.

Reply:
There are 4 patterns.

|setter|getter|desc|
|:---:|:---:|:---:|
| ×|×|the value works after get()|
| ○|×| setter has side effect |
| ×|○| this will no used |
| ○|○| setter/getter are just bypassing GObject property. |

Reply:
I see. Added a commit to reformat and store the scheme list.